dask_planner/src/expression.rs

```diff
@@ -180,9 +180,24 @@ impl PyExpr {
                     schema.merge(plan.schema().as_ref());
                 }
                 let name = get_expr_name(&self.expr).map_err(py_runtime_err)?;
-                schema
-                    .index_of_column(&Column::from_qualified_name(name))
-                    .map_err(py_runtime_err)
+                let index = schema
+                    .index_of_column(&Column::from_qualified_name(name.clone()))
+                    .map_err(py_runtime_err);
```

Comment:
Minor style preference. It might be possible for us to use `or_else` and instead map the error path to a function that implements the logic below.
Also we could split the logic below into a utility method in `parserUtils` that that returns the relation and table name, and use a closure here instead to map that result to a column and get the index

Reply:
Thanks! Lmk if the most recent changes are what you had in mind.

Reply:
Yeah this lgtm!